File: front-end/src/app/models/medical-prescription-dto-get.ts
import { PrescriptionStatus } from "./enums/prescription-status";

export interface MedicalPrescriptionDtoGet {
  id: number;
  medicationName: string;
  dosage: string;
  frequency: string;
  startDate: string;     
  endDate: string;         
  notes: string;
  prescribedBy: string;
  createdAt: string;  
  prescribedTo: string;
  clientEmail: string;
  status: string;     
}

File: back-end/src/main/java/com/tfg/back/model/dtos/medicalPrescription/MedicalPrescriptionDtoGet.java
package com.tfg.back.model.dtos.medicalPrescription;

import com.tfg.back.enums.PrescriptionStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class MedicalPrescriptionDtoGet {

    private Long id;

    private String medicationName;

    private String dosage;

    private String frequency;

    private LocalDate startDate;

    private LocalDate endDate;

    private String notes;

    private String prescribedBy;

    private LocalDateTime createdAt;

    private String prescribedTo;

    private String clientEmail;

    private PrescriptionStatus status;
}
